config_parser.cc

```diff
@@ -218,9 +228,16 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
     last_token_type = token_type;
   }
 
-  printf ("Bad transition from %s to %s\n",
-          TokenTypeAsString(last_token_type),
-          TokenTypeAsString(token_type));
+  if(bracket_count < 0) {
+      printf("Mismatched brackets found, extra \'}\' token(s)\n");
+  }
+  else if(bracket_count > 0) {
```

Comment:
If something other input breaks before the right bracket gets read, then this error will pop up, even though there is no mismatch.